nixos/modules/services/web-apps/tachidesk-server.nix

```
@@ -0,0 +1,227 @@
+          "server.socksProxyPort" = "8080";
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    networking.firewall.allowedTCPPorts = mkIf cfg.openFirewall [ cfg.config.port ];
+
+    users.groups = mkIf (cfg.group == "tachidesk") {
+      tachidesk = { };
+    };
+
+    users.users = mkIf (cfg.user == "tachidesk") {
+      tachidesk = {
+        group = cfg.group;
+        home = cfg.dataDir;
+        description = "Tachidesk Daemon user";
+        isSystemUser = true;
+      };
+    };
+
+    systemd.tmpfiles.rules = [
+      "d '${cfg.dataDir}/.local/share/Tachidesk' 0700 ${cfg.user} ${cfg.group} - -"
+    ];
+
+    systemd.services.tachidesk-server =
+      let
+        #  HOCON is a JSON superset that tachidesk-server use for configuration
+        toHOCON = attr: (
+          let
+            attrType = builtins.typeOf attr;
+          in
+          if builtins.elem attrType [ "string" "path" "int" "float" ] then ''"${toString attr}"''
+          else if attrType == "bool" then lib.boolToString attr
+          # attrs, list, lambda, null
+          else
+            throw ''
+              [tachidesk-server]: invalid config value type '${attrType}'.
+            ''
+        );
+        configFile = pkgs.writeText "server.conf" (lib.strings.concatStrings [
+          ''
+            server.systemTrayEnabled = "${lib.boolToString cfg.enableSystemTray}"
+
+            server.ip = "${cfg.config.ip}"
+            server.port = "${toString cfg.config.port}"
+
+            server.downloadAsCbz = ${lib.boolToString cfg.config.downloadAsCbz}
+            server.localSourcePath = "${cfg.config.localSourcePath}"
+
+            server.basicAuthEnabled = ${lib.boolToString cfg.config.basicAuth.enable}
+          ''
+
+          (
+            if cfg.config.basicAuth.enable then
+              (if builtins.isNull cfg.config.basicAuth.username || builtins.isNull cfg.config.basicAuth.passwordFile then
+                throw ''
+                  [tachidesk-server]: the username and the password file cannot be null when the basic auth is enable
+                '' else ''
+                server.basicAuthUsername = "${cfg.config.basicAuth.username}"
+                server.basicAuthPassword = "$TACHIDESK_SERVER_BASIC_AUTH_PASSWORD"
+              '') else ""
+          )
+
+          (lib.pipe cfg.extraConfig [
+            (lib.filterAttrs (_: x: x != null))
+            (lib.mapAttrsToList (name: value: ''${name} = ${toHOCON value}''))
+            lib.concatLines
+          ])
+        ]);
+      in
+      {
+        description = "A free and open source manga reader server that runs extensions built for Tachiyomi.";
+
+        wantedBy = [ "multi-user.target" ];
+        wants = [ "network-online.target" ];
+        after = [ "network-online.target" ];
+
+        environment = {
+          TACHIDESK_SERVER_BASIC_AUTH_PASSWORD = if cfg.config.basicAuth.enable then "${lib.fileContents cfg.config.basicAuth.passwordFile}" else "";
```

```suggestion
          TACHIDESK_SERVER_BASIC_AUTH_PASSWORD = lib.optionalString (!cfg.config.basicAuth.enable) "${lib.fileContents cfg.config.basicAuth.passwordFile}";
```
we have a shorthand for that

---

It's changed